Add a recipe for mcp-server-lib

laurynas-biveinis opened the pull request:
### Brief summary of what the package does

A library to write Model Context Protocol servers that run in Emacs.

### Direct link to the package repository

https://github.com/laurynas-biveinis/mcp-server-lib.el/

### Your association with the package

The maintainer

### Relevant communications with the upstream package maintainer

**None needed**

### Checklist

<!-- Please confirm by replacing `[]` with `[x]`: -->

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] I've used `M-x checkdoc` to check the package's documentation strings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)

~~There is one not fixed but disabled package-lint warning: the package provides short commands `mcp-start` and `mcp-stop` that do not start with the package name suffix. I feel that the package usability is much better with this, but if this is unacceptable, I will rename to `mcp-server-lib-start` and `mcp-server-lib-stop`.~~

<!-- After submitting, please fix any problems the CI reports. -->

laurynas-biveinis commented:
- Renamed mcp-start / mcp-stop to have proper package prefixes, removed the linter suppression
- Added a command to install the MCP stdio script into user-emacs-directory

riscy commented:
I just merged in another package called mcp: https://github.com/melpa/melpa/pull/9469#issuecomment-2907922991 - so there's a namespace conflict.  But do you know if there's an opportunity to team up here?  (Or maybe these are completely disjoint feature sets?)

In the meantime, quick first pass -

⸺ `mcp-server-lib-metrics.el` with package-lint 20250527.1845 and `package-lint-main-file` = "mcp-server-lib.el":
```
1 issue found:
7:0: error: Package-Requires outside the main file have no effect.
```

⸺ `mcp-server-lib-ert.el` with package-lint 20250527.1845 and `package-lint-main-file` = "mcp-server-lib.el":
```
1 issue found:
8:0: error: Package-Requires outside the main file have no effect.
```

⸺ `mcp-server-lib-ert.el` with [melpazoid](https://github.com/riscy/melpazoid):
```
- mcp-server-lib-ert.el#L45: This double negation can be collapsed (`not` aliases `null`)
```

⸺ `mcp-server-lib-commands.el` with [melpazoid](https://github.com/riscy/melpazoid):
```
- mcp-server-lib-commands.el#L147: Consider using %F instead of %Y-%m-%d in time strings
- mcp-server-lib-commands.el#L147: Consider using %T instead of %H:%M:%S in time strings
```

⸺ Package and license:
- Avoid `:commit` in recipes except in unusual cases
- Prefer equivalent recipe: `(mcp-server-lib :fetcher github :repo "laurynas-biveinis/mcp-server-lib.el" :commit "676df9937d40f0a03344628b662c13699c9724dd" :files (:defaults "emacs-mcp-stdio.sh" (:exclude "*-test*.el")))`
- mcp-server-lib.el must include all of the Package-Requires listed in mcp-server-lib-metrics.el, including: emacs "24.3"

Btw if you name your test files `test.el` I believe they'll be excluded by default.

riscy commented:
Disregard what I said about namespace conflict above. :)

laurynas-biveinis commented:
Thank you for the feedback!

I believe I have addressed everything:
- Added `package-lint-main-file` file locals where applicable
- Added melpazoid to CI, fixed the errors above
- Remove `:commit` from the recipe, using `:defaults` in `:files` to simplify, renamed mcp-server-lib-test-bytecode-handler.el to mcp-server-lib-bytecode-handler-test.el to remove `:exclude`

laurynas-biveinis commented:
@riscy, LGTM !

riscy commented:
Thanks!
